### DISCUSSION TITLE
hollow cynlinder geom

### DISCUSSION THREAD

**Discussion by shivanimall** [google-deepmind/mujoco#19] [Asking for Help] [answered]
Hello,

Is there an example to create hollow cylinder geom? Thank you!

**Comment by yuvaltassa** [comment, marked as answer]
I can think of 3 questions that you might be asking with 3 corresponding answers:

1. If you are asking about a hollow **collision** cylinder, no. In order to make concave collision geometries you need to form union of convex objects. This is fairly easy to do procedurally, e.g. by using our [mjcf](https://github.com/deepmind/dm_control/tree/master/dm_control/mjcf) package. Let us know if you need help with this.
2. If you are asking about a **visual** cylinder, (i.e. simply not rendering the two end-caps), we could potentially add such a [visual geom](https://github.com/deepmind/mujoco/blob/main/include/mjmodel.h#L115).
3. If you mean making a cylinder such that the **computed inertia** is concentrated on the geom’s shell, we are planning to add this option.

**Reply by shivanimall** [reply]
Many thanks @yuvaltassa for the detailed response, this is helpful! referring to (1) one. found an example for this: https://github.com/ARISE-Initiative/robosuite/blob/threading/robosuite/models/objects/composite/hollow_cylinder.py

**Reply by Krishnendu1984** [reply]
I want to create a prismatic joint with one cylinder going inside hollow cylinder how to do that .
Can anyone tell me

**Reply by Troldmand43** [reply]
I'm also looking for an answer to this question

**Reply by yuvaltassa** [reply]
The question is answered, but perhaps looking at the [mug model](https://github.com/google-deepmind/mujoco/blob/main/model/mug/mug.xml) would help.
